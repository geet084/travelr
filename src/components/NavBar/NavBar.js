import React, { Component } from 'react'
import NavBtn from '../NavBtn/NavBtn';
import Collapsible from 'react-collapsible';

export class NavBar extends Component {
  
  getCategoryTitles = () => {
    return Object.keys(this.props).map(key => {
      const updatedKey = key.split('');
      updatedKey[0] = updatedKey[0].toUpperCase();
      
      return updatedKey.join('');
    });
  }

  populateNavItems = (navCategories) => {
    return navCategories.map(title => {
      const pathname = title.toLowerCase();
      const navItemsInCategory = this.props[pathname].map(name => {
        return <NavBtn name={name} path={`/${pathname}/${name}`} key={pathname + name} />
      })

      return <Collapsible
        key={pathname}
        className="drop-down"
        trigger={`${title} ˅`}
        triggerWhenOpen={`${title} ˄`}
      >
        {navItemsInCategory}
      </Collapsible>;
    });
  }

  render() {
    const navCategoryTitles = this.getCategoryTitles();
    const navItems = this.populateNavItems(navCategoryTitles);

    return (
      <nav className="nav-bar">
        <ul>
          <NavBtn key={'home'} name={'Home'} path='/' />
          <Collapsible
            className="drop-down"
            trigger="Solar System ˅"
            triggerWhenOpen="Solar System ˄"
          >
            {navItems}
          </Collapsible>
        </ul>
      </nav>
    )
  }
}

export default NavBar;