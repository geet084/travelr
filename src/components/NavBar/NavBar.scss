ul {
  list-style-type: none;
  padding: 0 0 0 27px;
}

nav {
  font-size: calc(10px + 2vmin);
  color: white;
  position: fixed;
  top: 10%;
  width: 175px;
  border-radius: 7px;
}

li {
  &:hover {
    color: red;
  }
}

.nav-bar .Collapsible {
  color: white;
  &:hover {
    color: red;
  }
}

a {
  text-decoration: none;
  color: inherit;
  &:hover {
    color: blue;
  }
}

.nav-bar {
  color: white;
  background-color: rgba(255,255,255, 0.075);
  box-shadow: 0px 0px 5px 5px rgba(0,0,0,0.8);
}

.drop-down {
  color: lightblue;
}