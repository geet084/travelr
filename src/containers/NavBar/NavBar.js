import React, { Component } from 'react'
import NavBtn from '../../components/NavBtn/NavBtn'
import Collapsible from 'react-collapsible';

class NavBar extends Component {
  render() {
    const planets = ['Mercury', 'Venus', 'Earth', 'Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune']
    return (
      <nav>
        <ul>
          <NavBtn name={'Home'} path='/home'/>
          <NavBtn name={'Moon'} path='/moon'/>
          <Collapsible trigger="Planets">
            {planets.map(planet => <NavBtn name={planet} path={`/planets/${planet}`} key={planet} />)}
          </Collapsible>
          <NavBtn name={'Sun'} path='/sun'/>
        </ul>
      </nav>
    )
  }
}

export default NavBar;