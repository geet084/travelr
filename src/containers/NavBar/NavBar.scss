ul {
  list-style-type: none;
}

nav {
  font-size: calc(10px + 2vmin);
  color: white;
}

li {
  &:hover {
    color: red;
  }
}
.Collapsible {
  color: red;
}
