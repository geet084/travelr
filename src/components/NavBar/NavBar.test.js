import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

describe('NavBar', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      planets: ['name1', 'name2'],
      moons: ['name1', 'name2', 'name3'],
      stars: ['name8'],
      bodies: ['name5', 'name4'],
    }
    wrapper = shallow(<NavBar {...mockProps} />)
  });

  it('should match snapshot with no props', () => {
    const mockProps = {
      planets: [''],
      moons: [''],
      stars: [''],
      bodies: [''],
    }
    wrapper = shallow(<NavBar {...mockProps} />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshout with props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should get category title from props', () => {
    const expected = ["Planets", "Moons", "Stars", "Bodies"];
    const result = wrapper.instance().getCategoryTitles();

    expect(result).toEqual(expected);
  });

  it('should populate the nav bar with categories and items', () => {
    const moonIndex = Object.keys(mockProps).indexOf('moons');
    const moons = Object.keys(mockProps)[moonIndex]
    const expected = wrapper.instance().props[moons].length;

    const mockCategories = ["Planets", "Moons", "Stars", "Bodies"];
    
    const result = wrapper.instance().populateNavItems(mockCategories);
    
    expect(result[moonIndex].props.children).toHaveLength(expected);
  });
})