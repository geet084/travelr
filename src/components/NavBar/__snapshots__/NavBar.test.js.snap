// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot with no props 1`] = `
<nav
  className="nav-bar"
>
  <ul>
    <NavBtn
      key="home"
      name="Home"
      path="/"
    />
    <Collapsible
      className="drop-down"
      classParentString="Collapsible"
      contentContainerTagName="div"
      contentInnerClassName=""
      contentOuterClassName=""
      easing="linear"
      lazyRender={false}
      onClose={[Function]}
      onClosing={[Function]}
      onOpen={[Function]}
      onOpening={[Function]}
      open={false}
      openedClassName=""
      overflowWhenOpen="hidden"
      tabIndex={null}
      transitionCloseTime={null}
      transitionTime={400}
      trigger="Solar System ˅"
      triggerClassName=""
      triggerDisabled={false}
      triggerOpenedClassName=""
      triggerSibling={null}
      triggerStyle={null}
      triggerTagName="span"
      triggerWhenOpen="Solar System ˄"
    >
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="planets"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Planets ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Planets ˄"
      >
        <NavBtn
          key="planets"
          name=""
          path="/planets/"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="moons"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Moons ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Moons ˄"
      >
        <NavBtn
          key="moons"
          name=""
          path="/moons/"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="stars"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Stars ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Stars ˄"
      >
        <NavBtn
          key="stars"
          name=""
          path="/stars/"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="bodies"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Bodies ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Bodies ˄"
      >
        <NavBtn
          key="bodies"
          name=""
          path="/bodies/"
        />
      </Collapsible>
    </Collapsible>
  </ul>
</nav>
`;

exports[`NavBar should match snapshout with props 1`] = `
<nav
  className="nav-bar"
>
  <ul>
    <NavBtn
      key="home"
      name="Home"
      path="/"
    />
    <Collapsible
      className="drop-down"
      classParentString="Collapsible"
      contentContainerTagName="div"
      contentInnerClassName=""
      contentOuterClassName=""
      easing="linear"
      lazyRender={false}
      onClose={[Function]}
      onClosing={[Function]}
      onOpen={[Function]}
      onOpening={[Function]}
      open={false}
      openedClassName=""
      overflowWhenOpen="hidden"
      tabIndex={null}
      transitionCloseTime={null}
      transitionTime={400}
      trigger="Solar System ˅"
      triggerClassName=""
      triggerDisabled={false}
      triggerOpenedClassName=""
      triggerSibling={null}
      triggerStyle={null}
      triggerTagName="span"
      triggerWhenOpen="Solar System ˄"
    >
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="planets"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Planets ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Planets ˄"
      >
        <NavBtn
          key="planetsname1"
          name="name1"
          path="/planets/name1"
        />
        <NavBtn
          key="planetsname2"
          name="name2"
          path="/planets/name2"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="moons"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Moons ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Moons ˄"
      >
        <NavBtn
          key="moonsname1"
          name="name1"
          path="/moons/name1"
        />
        <NavBtn
          key="moonsname2"
          name="name2"
          path="/moons/name2"
        />
        <NavBtn
          key="moonsname3"
          name="name3"
          path="/moons/name3"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="stars"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Stars ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Stars ˄"
      >
        <NavBtn
          key="starsname8"
          name="name8"
          path="/stars/name8"
        />
      </Collapsible>
      <Collapsible
        className="drop-down"
        classParentString="Collapsible"
        contentContainerTagName="div"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        key="bodies"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger="Bodies ˅"
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={null}
        triggerTagName="span"
        triggerWhenOpen="Bodies ˄"
      >
        <NavBtn
          key="bodiesname5"
          name="name5"
          path="/bodies/name5"
        />
        <NavBtn
          key="bodiesname4"
          name="name4"
          path="/bodies/name4"
        />
      </Collapsible>
    </Collapsible>
  </ul>
</nav>
`;
